core/json/src/main/scala/net/liftweb/json/Formats.scala

```diff
@@ -230,9 +234,16 @@ case class ShortTypeHints(hints: List[Class[_]]) extends TypeHints {
 /** Use full class name as a type hint.
  */
 case class FullTypeHints(hints: List[Class[_]]) extends TypeHints {
+  private val hintsToClass: ConcurrentScalaMap[String, Class[_]] =
+    new ConcurrentHashMap[String, Class[_]]().asScala ++= hints.map(clazz => hintFor(clazz) -> clazz)
```

Comment:
It should be minimal for what we're using it for. Certainly we've seen a giant improvement from this change at Elemica. Doesn't mean that there isn't some performance left on the table, but compared to the big win we got from adding the caching it's negligible.